test_tools/log_parser/parser.py

```diff
@@ -110,19 +114,98 @@ def dump_event(evt, idx, is_dedup):
     else:
         print('unknown payload type')
 
-def dump_file(f):
-    buf = open(f, 'rb').read()
-    buf = bytearray(buf)
-    preamble = parse_preamble(buf)
 
-    batch = EventBatch.GetRootAsEventBatch(buf[PREAMBLE_LENGTH : PREAMBLE_LENGTH + preamble["msg_size"]], 0)
+def dump_event_batch(buf):
+    batch = EventBatch.GetRootAsEventBatch(buf, 0)
     meta = batch.Metadata()
-    print(f'parsed {f} with {batch.EventsLength()} events preamble:{preamble} enc:{meta.ContentEncoding()}')
-    is_dedup = b'ZSTD_AND_DEDUP' == meta.ContentEncoding()
+    enc = meta.ContentEncoding().decode('utf-8')
+    print(f'event-batch evt-count:{batch.EventsLength()} enc:{enc}')
+    is_dedup = b'DEDUP' == meta.ContentEncoding()
     for i in range(0, batch.EventsLength()):
-        dump_event(batch.Events(i), i, is_dedup)
+        dump_event(batch.Events(i), i)
     print("----\n")
 
+
+def dump_preamble_file(file_name, buf):
+    preamble = parse_preamble(buf)
+    print(f'parsing preamble file {file_name}\n\tpreamble:{preamble}')
+    dump_event_batch(buf[PREAMBLE_LENGTH : PREAMBLE_LENGTH + preamble["msg_size"]])
+
+MSG_TYPE_HEADER = 0x55555555
```

Comment:
Did this get checked in by mistake? The joined fb schema is missing

Reply:
yes, I checked in parser.py with partial support for the new file format :(